caproto/threading/client.py

```diff
@@ -1312,13 +1312,16 @@ def _process_command(self, command):
                                   ca.WriteNotifyResponse)):
             ioid_info = self.ioids.pop(command.ioid)
             deadline = ioid_info['deadline']
+            pv = ioid_info['pv']
             if deadline is not None and time.monotonic() > deadline:
-                self.log.warn("Ignoring late response with ioid=%d because "
+                self.log.warn("Ignoring late response with ioid=%d regarding "
+                              "PV named %s because "
                               "it arrived %.3f seconds after the deadline "
                               "specified by the timeout.", command.ioid,
-                              time.monotonic() - deadline)
+                              pv.name, time.monotonic() - deadline)
                 return
 
+            pv.log.debug("%r: %r", pv.name, command)
```

Comment:
Why `pv.log.debug(...)` is used here? Should it be `self.log.debug(...)` as above?

Reply:
@danielballan should verify, but this is so that you can look an the logging of just a single channel (aka PV) rather than all of the channels that are on the same circuit (aka IOC).

Reply:
Yes. The circuit logger (`self.log` here) shows everything that happens on a circuit, which is useful for debugging _caproto itself_ but often too much information for debugging hardware. The PV/channel loggers (`pv.log` here) are separate, and they let you focus the logging a little more.

In current master, we called `PV.log.debug` inside `PV.write()` and `PV.read()`. In this PR, we still `PV.log.debug`; we just do it from inside the circuit's receiving loop so that it also applies to batched requests and to writes that `PV.write()` is not waiting for.

Reply:
@danielballan, @tacaswell, thank you for the detailed explanation here and in a private channel! That answers my question.